arangod/RestHandler/RestAdminLogHandler.cpp

```diff
@@ -106,6 +124,51 @@ void RestAdminLogHandler::clearLogs() {
 }
 
 void RestAdminLogHandler::reportLogs(bool newFormat) {
+  bool foundServerIdParameter;
+  std::string serverId =
+      StringUtils::tolower(_request->value("serverId", foundServerIdParameter));
+
+  if (ServerState::instance()->isCoordinator() &&
+      ServerState::instance()->isRunningInCluster() && foundServerIdParameter) {
+    if (serverId != StringUtils::tolower(ServerState::instance()->getId())) {
+      // not ourselves! - need to pass through the request
+      auto& ci = server().getFeature<ClusterFeature>().clusterInfo();
+
+      bool found = false;
+      for (auto const& srv : ci.getServers()) {
+        // validate if server id exists
+        if (StringUtils::tolower(srv.first) == serverId) {
+          serverId = srv.first;
+          found = true;
+        }
+      }
+
+      if (!found) {
+        generateError(rest::ResponseCode::NOT_FOUND, TRI_ERROR_HTTP_BAD_PARAMETER,
+                      std::string("unknown serverId supplied.'"));
+        return;
+      }
+
+      NetworkFeature const& nf = server().getFeature<NetworkFeature>();
+      network::ConnectionPool* pool = nf.pool();
+      if (pool == nullptr) {
+        THROW_ARANGO_EXCEPTION(TRI_ERROR_SHUTTING_DOWN);
+      }
+
+      network::RequestOptions options;
+      options.timeout = network::Timeout(30.0);
```

Comment:
I think we should also set the database for the request here (as it will not be in `requestPath()`). Otherwise at least in theory there may be permission issues. Not that this would matter, because the API is restricted to admin users anyway, but it is good practice to issue the requests only inside the same database.